custom_metrics/privacy.js

```diff
@@ -15,10 +46,216 @@ return JSON.stringify({
+        if (success) {
+          consentData = result;
+          consentData.compliant_setup = true;
+        } else {
+          // special case for consentmanager ('CMP settings are used that are not compliant with the IAB TCF')
+          // see warning at the top of https://help.consentmanager.net/books/cmp/page/changes-to-the-iab-cmp-framework-js-api
+          // cf. https://help.consentmanager.net/books/cmp/page/javascript-api
+          // Test site: https://www.pokellector.com/
+          __cmp('noncompliant_getVendorConsents', null, (result, success) => {
+            if (success) {
+              consentData = result;
+              consentData.compliant_setup = false;
+            } else {
+              consentData = {error: 'Failed to retrieve consent data'};
+            }
+          });
+        }
+      });
+
+      // TODO: could improve to collect 'consent string' data (see next line), but I haven't been able to find a website that still uses v1 to test this
+      // __cmp("getConsentData", null, function(v, success) { console.log(v); })
+    } else {
+      consentData = null;
+    }
+    return consentData;
+  })(),
+
+  /**
+   * IAB Transparency and Consent Framework v2
+   * docs v2: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2
+   *
+   * Test site: https://www.rtl.de/
+   */
+  iab_tcf_v2: (() => {
+    let tcData;
+    // description of `__tcfapi`: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-provide-the-api
+    if (typeof window.__tcfapi == 'function') {
+      // based on https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#gettcdata
+      __tcfapi('getTCData', 2, (result, success) => {
+        if (success) {
+          tcData = result;
+          tcData.compliant_setup = true;
+        } else {
+          // special case for consentmanager ('CMP settings are used that are not compliant with the IAB TCF')
+          // see warning at the top of https://help.consentmanager.net/books/cmp/page/changes-to-the-iab-cmp-framework-js-api
+          // cf. https://help.consentmanager.net/books/cmp/page/javascript-api
+          // Test site: ?
+          __tcfapi('noncompliant_getTCData', 2, (result, success) => {
+            if (success) {
+              tcData = result;
+              tcData.compliant_setup = false;
+            } else {
+              tcData = {error: 'Failed to retrieve TCData'};
+            }
+          });
+        }
+      });
+    } else {
+      tcData = null;
+    }
+    return tcData;
+  })(),
+
+  /**
+   * IAB US Privacy User Signal Mechanism “USP API”
+   * https://github.com/InteractiveAdvertisingBureau/USPrivacy
+   *
+   * Test site: https://www.nfl.com/
+   */
+  iab_usp: (() => {
```

Comment:
@VictorLeP I suggest rewriting tcf detection similar to this version of `iab_usp`.
After some testing, I identified that the previous version of `iab_usp` on nfl.com resulted in false-negative result.

```javascript
(() => {
    let uspData;
    if (typeof __uspapi == 'function') {
      __uspapi('getUSPData', 1, (result, success) => {
        if(success) {
          uspData = result;
        } else {
          uspData = {"error": "Failed to retrieve USPData"};
        }
      });
    } else {
      uspData = null;
    }
    return uspData;
  })()
```
> undefined

as compared to the new one:
```javascript
(() => {
    let uspData = {
      present: typeof window.__uspapi == 'function',
      privacy_string: null,
    };
    if (uspData.present) {
      window.__uspapi('getUSPData', 1, (result, success) => {
        if (success) {
          uspData.privacy_string = result;
        }
      });
    }
    return uspData;
  })()
```
> {present: true, privacy_string: null}

What do you think?

Reply:
If it only works correctly in the rewritten version, then of course we should use that one :)
I'll look into rewriting the TCF checks in a similar way.

Reply:
Refactor done